src/execution/index/unbound_index.cpp

```diff
@@ -0,0 +1,35 @@
+#include "duckdb/execution/index/unbound_index.hpp"
+#include "duckdb/parser/parsed_data/create_index_info.hpp"
+#include "duckdb/storage/table_io_manager.hpp"
+#include "duckdb/storage/block_manager.hpp"
+#include "duckdb/storage/index_storage_info.hpp"
+
+namespace duckdb {
+
+//-------------------------------------------------------------------------------
+// Unbound index
+//-------------------------------------------------------------------------------
+
+UnboundIndex::UnboundIndex(const CreateIndexInfo &info, IndexStorageInfo storage_info_p,
+                           TableIOManager &table_io_manager, AttachedDatabase &db)
+    : Index(info.index_name, info.index_type, info.constraint_type, info.column_ids, table_io_manager, db),
+      create_info(info), storage_info(std::move(storage_info_p)) {
+
+	// Annoyingly, parsed expressions are not copied in the CreateIndexInfo copy constructor
+	for (auto &expr : info.parsed_expressions) {
```

Comment:
```c++
unique_ptr<CreateInfo> CreateIndexInfo::Copy() const {

	auto result = make_uniq<CreateIndexInfo>(*this);
	CopyProperties(*result);

	for (auto &expr : expressions) {
		result->expressions.push_back(expr->Copy());
	}
	for (auto &expr : parsed_expressions) {
		result->parsed_expressions.push_back(expr->Copy());
	}

	return std::move(result);
}
```
?

Reply:
Annoyingly, thats not a copy-constructor

Reply:
But yes, should probably use that instead

Reply:
We intentionally don't use copy constructors for more complex/non-trivial types since copy constructors are rather error-prone, and instead use a dedicated `Copy()` method to make the intention more clear